_resources/port1.0/group/gnuradio-1.0.tcl

```diff
@@ -0,0 +1,201 @@
+    set py_ver_no_dot [join [split ${py_ver} "."] ""]
+    if {[variant_isset python${py_ver_no_dot}]} {
+        set active_python_version        ${py_ver}
+        set active_python_version_no_dot ${py_ver_no_dot}
+    }
+}
+
+depends_build-append \
+    port:pkgconfig
+
+depends_lib-append \
+    port:fftw-3-single \
+    path:lib/libmpir.dylib:mpir \
+    path:lib/libvolk.dylib:volk \
+    port:gmp \
+    port:python${active_python_version_no_dot}
+
+# need matplotlib for polar encoder/decoder, runtime only
+depends_run-append \
+    port:py${active_python_version_no_dot}-matplotlib \
+    port:py${active_python_version_no_dot}-opengl \
+    port:py${active_python_version_no_dot}-scipy \
+    port:py${active_python_version_no_dot}-numpy \
+    port:py${active_python_version_no_dot}-cheetah
+
+if {[string first "37" $subport] > 1} {
+    # still require cppunit for testing; NOTE: cppunit is checked for
+    # during configure, so we need it to be in depends_lib or
+    # depends_build to be used correctly. Choose the latter since it's
+    # not required for runtime; just for build/test.
+    depends_build-append \
+        port:cppunit
+} else {
+    # gr-vocoder removed in-tree libgsm, libcodec2, use system-wide libs
+    depends_lib-append \
+        port:codec2 \
+        port:libgsm
+
+    # gr_modtool requires click and click-plugins, and these are
+    # checked for at configure time, so have to use 'lib' here
+    depends_lib-append \
+        port:py${active_python_version_no_dot}-click \
+        port:py${active_python_version_no_dot}-click-plugins
+}
+
+if {[string first "gr37-" $subport] >= 0} {
+    depends_lib-append \
+        port:gnuradio37
+
+    depends_build-append \
+        port:swig3-python
+
+    configure.args-append \
+        -DSWIG_EXECUTABLE=${prefix}/bin/swig3
+} elseif {[string first "gr-" $subport] >= 0} {
+    depends_lib-append \
+        path:lib/libgnuradio-runtime.dylib:gnuradio
+
+    depends_build-append \
+        port:swig-python
+
+    configure.args-append \
+        -DSWIG_EXECUTABLE=${prefix}/bin/swig
+}
+
+# set build type to release or debug, not "MacPorts" since the GR
+# CMake system checks for a valid value and this is not one of them
+if {[variant_isset debug]} {
+    cmake.build_type Debug
+} else {
+    cmake.build_type Release
+}
+
+# specify the Python version to use
+set python_framework_dir ${frameworks_dir}/Python.framework/Versions/${active_python_version}
+configure.args-append \
+    -DPYTHON_EXECUTABLE=${python_framework_dir}/bin/python${active_python_version} \
+    -DPYTHON_INCLUDE_DIR=${python_framework_dir}/Headers \
+    -DPYTHON_LIBRARY=${python_framework_dir}/Python \
+    -DGR_PYTHON_DIR=${python_framework_dir}/lib/python${active_python_version}/site-packages \
+    -DDOXYGEN_DOT_EXECUTABLE=${prefix}/bin/dot \
+    -DDOXYGEN_EXECUTABLE=${prefix}/bin/doxygen \
+    -DCMAKE_MODULES_DIR=${prefix}/share/cmake \
+    -DGMP_INCLUDE_DIR= \
```

Comment:
Technically I think these CMake defines  (GMS and MPIR) belong in the GR Portfile, not here, since they are not common to all GR OOT modules. That said these won't hurt either since if CMake doesn't need them then they will just be ignored.

Reply:
I use the same considerations said before. Stay here. It generalizes the portgroup for every necessity.